django/template/defaulttags.py

```diff
@@ -36,15 +36,22 @@ def __init__(self, setting, nodelist):
         self.setting, self.nodelist = setting, nodelist
 
     def render(self, context):
-        old_setting = context.autoescape
-        context.autoescape = self.setting
-        output = self.nodelist.render(context)
-        context.autoescape = old_setting
+        output = ''.join(self.stream(context))
         if self.setting:
             return mark_safe(output)
         else:
             return output
 
+    def stream(self, context):
+        old_setting = context.autoescape
+        context.autoescape = self.setting
+        for chunk in self.nodelist.stream(context):
+            if self.setting:
+                yield mark_safe(chunk)
```

Comment:
I can't be sure it is performance critical without testing but it looked like it could be when dealing with large loops. I just suggested it because a ticket was recently opened about a template performance regression in between 1.7.x and 1.8.x if I remember correctly.